components/region_cache_memory_engine/src/background.rs

```diff
@@ -99,6 +109,71 @@ impl Drop for BgWorkManager {
     }
 }
 
+pub struct PdRangeHintService(Arc<RpcClient>);
+
+impl RangeHintService for PdRangeHintService {}
+
+impl From<Arc<RpcClient>> for PdRangeHintService {
+    fn from(pd_client: Arc<RpcClient>) -> Self {
+        PdRangeHintService(pd_client)
+    }
+}
+
+const CACHE_LABEL_RULE_KEY: &str = "cache";
+const CACHE_LABEL_RULE_ALWAYS: &str = "always";
+
+impl PdRangeHintService {
+    pub fn start(
+        &self,
+        scheduler: Scheduler<BackgroundTask>,
+        remote: Remote<yatp::task::future::TaskCell>,
+        load_and_prepare: LoadAndPrepareRangeFn,
+    ) {
+        let pd_client = self.0.clone();
+        let region_label_added_cb: RegionLabelAddedCb = Arc::new(move |label_rule: LabelRule| {
+            if !label_rule
+                .labels
+                .iter()
+                .any(|e| e.key == CACHE_LABEL_RULE_KEY && e.value == CACHE_LABEL_RULE_ALWAYS)
+            {
+                // not related to caching, skip.
+                return;
+            }
+            for key_range in &label_rule.data {
+                match CacheRange::try_from(key_range) {
+                    Ok(cache_range) => {
+                        info!("Loading range"; "cache_range" => ?&cache_range);
+                        if let Err(reason) = load_and_prepare(&cache_range) {
+                            error!("Load and prepare failed"; "range" => ?&cache_range, "reason" => ?reason);
+                        } else {
+                            scheduler.schedule(BackgroundTask::LoadRange).unwrap();
```

Comment:
It seems to me multiple LoadRange task could run concurrently and may lead to duplicated region load?

Reply:
Spoke offline about this. Will make this change.